crates/oxc_linter/src/snapshots/consistent_type_definitions.snap

```diff
@@ -0,0 +1,104 @@
+---
+source: crates/oxc_linter/src/tester.rs
+expression: consistent_type_definitions
+---
+  ⚠ typescript-eslint(consistent-type-definitions):
+   ╭─[consistent_type_definitions.tsx:2:28]
+ 1 │     
+ 2 │ ╭─▶         export type W<T> = {
+ 3 │ │             x: T;
+ 4 │ ├─▶         };
+   · ╰──── Use an `interface` instead of a `type`
+ 5 │                   
+   ╰────
+  help: Use an `interface` instead of a `type`
+
+  ⚠ typescript-eslint(consistent-type-definitions):
+   ╭─[consistent_type_definitions.tsx:2:16]
+ 1 │     
+ 2 │ ╭─▶         export interface W<T> {
+ 3 │ │             x: T;
+ 4 │ ├─▶         }
+   · ╰──── Use an `type` instead of a `interface`
+ 5 │                   
+   ╰────
+  help: Use an `type` instead of a `interface`
+
+  ⚠ typescript-eslint(consistent-type-definitions):
+   ╭─[consistent_type_definitions.tsx:3:11]
+ 2 │             namespace JSX {
+ 3 │ ╭─▶           interface Array<T> {
```

Comment:
Shall we only point the keyword `interface` or `type` in to make the error more specific?

Reply:
Some of the hints pointed to the entire expression, when the error is for a single line type, and to the stuff between the brackets, when multiline. I've updated it to always point to the entire expression in the name of consistency.

I have no strong preference where the error should point tho (and I couldn't find how to get the `span` of the `interface`/`type` keyword).

Interested to hear what you think :)

Reply:
For `TSInterfaceDeclaration`, the keyword `interface`'s span is from `TSInterfaceDeclaration.start` to `TSInterfaceDeclaration.start + 9` (9 is the length of 'interface'), `type` in `TSTypeAliasDeclaration` is the same as it.

For `ExportDefaultDeclaration`, looks like we can iterate source text from `ExportDefaultDeclaration.start`, and find the target keyword? 

```rust
let slice = ctx.semantic().source_text()[(start.start as usize)..].chars();
// ...
```

Reply:
just learned something new, thanks!